exp/internal/controllers/machinepool_controller_phases.go

```diff
@@ -274,6 +278,10 @@ func (r *MachinePoolReconciler) reconcileInfrastructure(ctx context.Context, clu
 		conditions.WithFallbackValue(ready, clusterv1.WaitingForInfrastructureFallbackReason, clusterv1.ConditionSeverityInfo, ""),
 	)
 
+	if err := r.reconcileMachines(ctx, mp, infraConfig); err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "failed to reconcile MachinePool machines for MachinePool `%s`", mp.Name)
```

Comment:
```suggestion
		return ctrl.Result{}, errors.Wrapf(err, "failed to reconcile machines for MachinePool `%s`", mp.Name)
```
nit (it is already a message for the MP controller, so it is implicit)